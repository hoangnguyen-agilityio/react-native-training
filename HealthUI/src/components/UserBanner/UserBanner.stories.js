import { storiesOf } from '@storybook/react-native';
import React from 'react';
import UserBanner from '.';

storiesOf('UserBanner', module)
  .add('default', () => (
    <UserBanner
      title="Hi, Nam Duong"
      date="Wednesday 29 Dec"
      avatarUrl={require("../../assets/user.png")}
    />
  ));
