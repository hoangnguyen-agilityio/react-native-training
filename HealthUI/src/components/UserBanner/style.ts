import { StyleSheet } from 'react-native';
import colors from '../../constants/colors';

const styles = StyleSheet.create({
  root: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
  },
  title: {
    fontFamily: 'SFPRODISPLAYBOLD',
    fontSize: 24,
    color: colors.dark,
  },
  date: {
    fontFamily: 'SFPRODISPLAYMEDIUM',
    color: colors.purple,
    fontSize: 16,
  },
  avatar: {
    width: 44,
    height: 44,
  },
});

export default styles;
