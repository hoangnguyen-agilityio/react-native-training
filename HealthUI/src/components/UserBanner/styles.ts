import { StyleSheet } from 'react-native';
import colors from '../../constants/colors';

const styles = StyleSheet.create({
  root: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
  },
  title: {
    fontFamily: 'SF-Pro-Display-Bold',
    fontSize: 24,
    color: colors.dark,
  },
  date: {
    fontFamily: 'SF-Pro-Display-Medium',
    color: colors.purple,
    fontSize: 16,
  },
  avatar: {
    width: 44,
    height: 44,
  },
});

export default styles;
