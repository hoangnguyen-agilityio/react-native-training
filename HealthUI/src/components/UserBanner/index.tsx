import React from 'react';
import { View, Image, Text, ImageSourcePropType } from 'react-native';
import styles from './styles';

interface Props {
  title: string;
  date: string;
  avatarUrl: ImageSourcePropType;
}

const UserBanner = (props: Props) => {
  const { title, date, avatarUrl } = props;

  return (
    <View style={styles.root}>
      <View>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.date}>{date}</Text>
      </View>

      <Image style={styles.avatar} source={avatarUrl} />
    </View>
  );
};

export default UserBanner;
