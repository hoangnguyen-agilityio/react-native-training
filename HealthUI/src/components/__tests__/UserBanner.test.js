import React from 'react';
import { Image, Text } from 'react-native';
import renderer from 'react-test-renderer';
import UserBanner from '../UserBanner';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <UserBanner
        title="Hi, Nam Duong"
        date="Wednesday 29 Dec"
        avatarUrl={require('../../assets/user.png')}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('component elements', () => {
  const tree = renderer.create(
    <UserBanner
      title="Hi, Nam Duong"
      date="Wednesday 29 Dec"
      avatarUrl={require('../../assets/user.png')}
    />,
  ).root;

  expect(tree.findAllByType(Text)).toHaveLength(2);
  expect(tree.findAllByType(Image)).toHaveLength(1);
});
