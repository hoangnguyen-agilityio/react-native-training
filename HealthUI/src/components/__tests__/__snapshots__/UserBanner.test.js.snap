// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#293032",
          "fontFamily": "SFPRODISPLAYBOLD",
          "fontSize": 24,
        }
      }
    >
      Hi, Nam Duong
    </Text>
    <Text
      style={
        Object {
          "color": "#5142AB",
          "fontFamily": "SFPRODISPLAYMEDIUM",
          "fontSize": 16,
        }
      }
    >
      Wednesday 29 Dec
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/user.png",
      }
    }
    style={
      Object {
        "height": 44,
        "width": 44,
      }
    }
  />
</View>
`;
